src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { ReactElement, useState, useEffect } from 'react';
+import {
+  Button,
+  Menu,
+  MenuItem,
+  Dialog,
+  DialogTitle,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  ListItemText,
+  IconButton,
+  styled,
+  TextareaAutosize,
+} from '@material-ui/core';
+import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
+import CloseIcon from '@material-ui/icons/Close';
+import { useTranslation } from 'react-i18next';
+import { DateTime } from 'luxon';
+import { Skeleton } from '@material-ui/lab';
+import {
+  useGetTaskAnalyticsQuery,
+  useGetEmailNewsletterContactsLazyQuery,
+} from '../NewsletterMenu.generated';
+
+interface Props {
+  accountListId: string;
+}
+
+const CloseButton = styled(IconButton)(({ theme }) => ({
+  position: 'absolute',
+  right: theme.spacing(1),
+  top: theme.spacing(1),
+  color: theme.palette.text.primary,
+  '&:hover': {
+    backgroundColor: '#EBECEC',
+  },
+}));
+
+const NewsletterTextContainer = styled(ListItemText)(({}) => ({
+  textAlign: 'left',
+}));
+
+const TextArea = styled(TextareaAutosize)(({}) => ({
+  width: '100%',
+}));
+
+const NewsletterMenuDialogTitle = styled(DialogTitle)(({}) => ({
+  textTransform: 'uppercase',
+  textAlign: 'center',
+}));
+
+const NewsletterMenu = ({ accountListId }: Props): ReactElement<Props> => {
+  const { t } = useTranslation();
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedMenuItem, changeSelectedMenuItem] = useState(-1);
+  const [newsletterMenuDialogOpen, changeNewsletterMenuDialogOpen] = useState(
+    false,
+  );
+  const [emailList, changeEmailList] = useState('');
+
+  const {
+    data: {
+      taskAnalytics: {
+        lastElectronicNewsletterCompletedAt,
+        lastPhysicalNewsletterCompletedAt,
+      } = {},
+    } = {},
+    loading,
+  } = useGetTaskAnalyticsQuery({
+    variables: { accountListId },
+  });
+
+  const [
+    loadEmailNewsletterContacts,
+    { data: newsletterContactsData, loading: contactsLoading },
+  ] = useGetEmailNewsletterContactsLazyQuery({
+    variables: { accountListId },
+  });
+
+  useEffect(() => {
+    if (newsletterContactsData?.contacts.nodes.length > 0) {
+      changeEmailList(
+        newsletterContactsData.contacts.nodes.reduce(
```

Comment:
a `.map` to get an array of emails and a `.join(',')` might be cleaner than the reduce you have. Might have to throw a `.filter(Boolean)` in there if you need null checks. 🤷‍♂️

Reply:
Yep, I like that better 🙂